src/psyclone/psyGen.py

```diff
@@ -4241,9 +4218,28 @@ def access(self):
 
     @access.setter
     def access(self, value):
-        ''' set the access type for this argument '''
+        '''Set the access type for this argument.
+        :param value: New access type.
+        :type value: :py:class:`psyclone.core.access_type`.
+        :raises ValueError if value is not an AccessType.
+        '''
+        if not isinstance(value, AccessType):
+            raise ValueError("Invalid access type: '{0}'.".format(value))
+
         self._access = value
 
+    def is_read(self):
+        '''Checks if the argument is read, i.e. has access
+        type read, readwrite, or inc.'''
+        return self._access in [AccessType.READ, AccessType.READWRITE,
```

Comment:
`in self._read_access_types` perhaps?

Reply:
The two is_read/write functions are actually a left over before I decided to use enums. So I have removed the completely.